### PR TITLE
Fix cli insufficient balance error message

### PR DESCRIPTION
**Pull request by iketheadore**
Differ insufficient balance in wallet and insufficient confirmed balance, cli should give different message to indicate whether the wallet does really not have sufficient balance, or just the balance have not confirmed yet.

### CODE REVIEW

File: src/api/cli/create_rawtx.go [hunk 1]
```diff
@@ -430,36 +424,36 @@ func getKeys(wlt *wallet.Wallet, outs []UnspentOut) ([]cipher.SecKey, error) {
...
+	unconfirmedBalance, err := unconfirmedOuts.Balance()
+	if err != nil {
+		return nil, fmt.Errorf("get unconfirmed balance failed: %v", err)
+	}
 
-			if totalCoins >= coins {
-				return outs, nil
-			}
-		}
+	if unconfirmedBalance.Coins+spendCoins < coins {
```
Comment (gz-c) [inline src/api/cli/create_rawtx.go]: `if unconfirmedBalance.Coins+spendCoins > coins` ?

Comment (iketheadore) [reply src/api/cli/create_rawtx.go]: oh, should rename the unconfirmedBalance to unconfirmedIncoming, and spendCoins to spendableCoins

Comment (gz-c) [reply src/api/cli/create_rawtx.go]: Can you fix it, the code seems wrong

File: src/api/cli/create_rawtx.go [hunk 2]
```diff
@@ -430,36 +424,39 @@ func getKeys(wlt *wallet.Wallet, outs []UnspentOut) ([]cipher.SecKey, error) {
...
+	uncfmIncoming, err := uncfmIncomingOuts.Balance()
+	if err != nil {
+		return nil, fmt.Errorf("get unconfirmed balance failed: %v", err)
+	}
 
-			if totalCoins >= coins {
-				return outs, nil
-			}
-		}
+	if spendableCoins+uncfmIncoming.Coins < coins {
```
Comment (gz-c) [inline src/api/cli/create_rawtx.go]: Are you sure these two errors are not in the wrong order?

Comment (iketheadore) [reply src/api/cli/create_rawtx.go]: Yes, if the spendable coins plus the unconfirmed incoming coins still less than the coins you want to spend, that means you don't have enough coins in wallet.

**Pull request merged**